### PR TITLE
feat: sign forc-deploy transactions

### LINKED ISSUE #2623: Refactor `forc-deploy` so that it returns an transaction ID and asks for a signature
As we are moving forward with #2413, we need `forc-deploy` to report the transaction id (deployment transaction) back to the user and then wait for the signature. A signature will be generated by the `forc-wallet` as we sign the transaction with that. After the signature is generated we need to give it back to `forc-deploy`. 

Also, it is important that (maybe using a flag) we should be able to keep the current behavior as testing suite depends on that

## Blocked by:

- [x] https://github.com/FuelLabs/fuels-rs/pull/540 (need a release which includes this)
- [x] https://github.com/FuelLabs/forc-wallet/pull/24

### PR DESCRIPTION
**Pull request by kayagokalp**
## About this PR
closes #2623.

~Blocked until https://github.com/FuelLabs/fuels-rs/pull/540 is merged and released. While testing I pointed to a local instance of the SDK,~ most of the diff is actually coming from the lock file.

### TODO

- [x] Accept transaction parameters 
- [x] Point to released fuels-rs (Blocked until the next release of fuels-rs)

### CODE REVIEW

**Comment by mitchmindtree** [comment]
Just adding a note (previously discussed on slack with @kayagokalp):

I think it might be best if we first wait to update `fuel-core`, `fuels-rs` and their dependencies that we use throughout the Sway repo before this lands. We want to avoid the massive lock file additions this PR would otherwise cause, which would require anyone building the repo to build both entire version sets and likely slow down CI quite a bit.

https://github.com/FuelLabs/sway/issues/2622 tracks the `fuel-core` update blockers in particular.

**Comment by mitchmindtree** [comment]
Ahh it looks like https://github.com/FuelLabs/sway/pull/2482 is doing the heavy-work of updating to the latest `fuel-core` and `fuels-rs`.

File: forc-plugins/forc-client/src/ops/deploy/cmd.rs [hunk 1]
```diff
@@ -69,4 +69,13 @@ pub struct DeployCommand {
     /// Output the time elapsed over each part of the compilation process.
     #[clap(long)]
     pub time_phases: bool,
+    /// Do not sign the transaction
+    #[clap(long)]
+    pub not_sign: bool,
```
Comment (mitchmindtree) [inline forc-plugins/forc-client/src/ops/deploy/cmd.rs:74]: Was just taking a quick look at #2656 and noticed this flag - perhaps a better name might be `unsigned`?

**Review by mitchmindtree — DISMISSED** [review]
:rocket: 

Looking great! Just a couple conflicts to resolve.

**Review by sezna — DISMISSED** [review]

**Review by sezna — APPROVED** [review]

**Review by mitchmindtree — APPROVED** [review]

**Pull request merged**